mono/mini/intrinsics.c

```diff
@@ -543,13 +543,7 @@ emit_unsafe_intrinsics (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSignatu
 
 		t = ctx->method_inst->type_argv [0];
 		t = mini_get_underlying_type (t);
-		if (MONO_TYPE_IS_PRIMITIVE (t) && t->type != MONO_TYPE_R4 && t->type != MONO_TYPE_R8) {
```

Comment:
emit_memory_load () will emit a memcpy for this, is that ok ?

Reply:
I think that's OK. `mini_emit_memory_load` will emit a memcpy for this when not using LLVM as a backend, exactly the same as is already done when using the [`unaligned` prefix](https://github.com/imhameed/mono/blob/df456a935c11ffc0c5d4082847a7e85d7d3252cc/mono/mini/method-to-ir.c#L11246) with an [`ldind`](https://github.com/imhameed/mono/blob/df456a935c11ffc0c5d4082847a7e85d7d3252cc/mono/mini/method-to-ir.c#L8296) or [`ldobj`](https://github.com/imhameed/mono/blob/df456a935c11ffc0c5d4082847a7e85d7d3252cc/mono/mini/method-to-ir.c#L8546). In the case that `t->type` equals `MONO_TYPE_R4` or `MONO_TYPE_R8`, [an `unaligned.ldobj` sequence ](https://github.com/mono/corefx/blob/b9e66087d6a9d4eb3a30c80ead3b8402510dad3d/src/System.Runtime.CompilerServices.Unsafe/src/System.Runtime.CompilerServices.Unsafe.il#L66) would have been inlined into the calling function anyway, so there wouldn't be any change in behavior there.

If you're concerned about these intrinsics emitting a memcpy on targets like i386, amd64, or arm64, where unaligned memory access is cheap and transparent, I could change this PR to also introduce some more target-specific tweaks, and then other CIL opcodes that can support the `unaligned` prefix would benefit, in addition to these intrinsics.